compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala

```diff
@@ -87,8 +87,15 @@ final class Conversions(
         setCrash(reason)
       case SError.DamlEArithmeticError(reason) =>
         setCrash(reason)
-      case SError.DamlEUnhandledException(_, exc) =>
-        builder.setUnhandledException(convertValue(exc))
+      case SError.DamlEUnhandledException(exc) =>
+        exc match {
+          case SValue.SAnyException(_, sValue) =>
+            builder.setUnhandledException(convertValue(sValue.toValue))
+          case SValue.SBuiltinException(SValue.ArithmeticError) =>
+            setCrash("ArithmeticError")
+          case SValue.SBuiltinException(SValue.ContractError) =>
+            setCrash("ContractError")
```

Comment:
Please add TODOs here. We shouldn’t crash.

Reply:
6b620b826b5b187795aa62f7503cb7de7996d9e8